code/game/objects/items/weapons/tools.dm

```diff
@@ -317,17 +317,26 @@
 		remove_fuel(1)
 		var/turf/location = get_turf(user)
 		location.hotspot_expose(700, 50, 1)
+		if(get_fuel() <= 0)
+			user.AddLuminosity(-light_intensity)
 
 		if(isliving(O))
 			var/mob/living/L = O
 			if(L.IgniteMob())
 				message_admins("[key_name_admin(user)] set [key_name_admin(L)] on fire")
 				log_game("[key_name(user)] set [key_name(L)] on fire")
 
+
 /obj/item/weapon/weldingtool/attack_self(mob/user)
-	toggle(user)
+	switched_on(user)
+	if(welding)
+		SetLuminosity(0)

```

Comment:
Yeah, they still need to be there. Without those lines of code if you picked up a welder that was already turned on it wouldn't illuminate the player.